CMakeLists.txt

```diff
@@ -113,21 +103,95 @@ endif()
 
 add_library(${PROJECT_NAME} ${SOURCES})
 
-set_target_properties(${MOVEIT_LIB_NAME} PROPERTIES VERSION ${${PROJECT_NAME}_VERSION})
+target_include_directories(${PROJECT_NAME} PUBLIC
+  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+  $<INSTALL_INTERFACE:include>
+)
 
-add_dependencies(${PROJECT_NAME} ${${PROJECT_NAME}_EXPORTED_TARGETS} ${catkin_EXPORTED_TARGETS})
+ament_target_dependencies(
+  ${PROJECT_NAME}
+  PUBLIC
+    Eigen3
+    moveit_core
+    moveit_ros_planning
+    pluginlib
+    rclcpp
+    tf2_eigen
+    tf2_ros
+    tf2_kdl
+    tf2_geometry_msgs
+)
 
-target_link_libraries(${PROJECT_NAME}
-  ${catkin_LIBRARIES}
+target_link_libraries(
+  ${PROJECT_NAME}
+  PUBLIC
   ${FANN_LIBRARIES}
   ${OpenMP_LIBS}
 
   -static-libgcc
   -static-libstdc++
 )
 
-install(DIRECTORY include/${PROJECT_NAME}/ DESTINATION ${CATKIN_PACKAGE_INCLUDE_DESTINATION})
+add_library(${PROJECT_NAME}_plugin SHARED
+  ${SOURCES}
+  src/kinematics_plugin.cpp
+)
 
-install(TARGETS ${PROJECT_NAME} LIBRARY DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION})
+target_include_directories(${PROJECT_NAME}_plugin PUBLIC
+  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+  $<INSTALL_INTERFACE:include>
+)
+
+target_link_libraries(
+  ${PROJECT_NAME}_plugin
+  PRIVATE
+    bio_ik
+
+    -static-libgcc
+    -static-libstdc++
```

Comment:
This is redundant because these are already specified as PUBLIC link libraries of the core `bio_ik` target. By the way, to match the style of the rest of this CMakeLists, you may want to change `bio_ik` to `${PROJECT_NAME}`. In general I don't recommend this practice, but that's outside of the scope of this PR which ought to match the existing style.